### ISSUE TITLE
TextInput focus shows status bar permanently on full screen apps.

### ISSUE THREAD

**Issue by sebastiantoro** [BowlerHatLLC/feathers#1597]
I'm using Starling 2.1 and Feathers 3.1.2 and found a bug related to the TextInput component. When you focus on the TextInput the Android' softkeyboard and status bar appear. When you focus outside the TextInput compoenent, only the softkeyboard goes away leaving the status bar visible.

To replicate:

```
[SWF(width="1024", height="600", frameRate="60", backgroundColor="#ffffff")]
	public class TextInputBug extends Sprite
	{
		private var _starling:Starling;
 
		public function TextInputBug()
		{
			NativeApplication.nativeApplication.systemIdleMode = SystemIdleMode.KEEP_AWAKE;
 
			this.stage.scaleMode = StageScaleMode.NO_SCALE;
			this.stage.align = StageAlign.TOP_LEFT;
 
			this.loaderInfo.addEventListener(Event.COMPLETE, loaderInfoCompleteHandler);
		}
 
		protected function loaderInfoCompleteHandler(event:Event):void
		{
			_starling = new Starling(AppManager, stage);
			_starling.start();
		}
	}
```

```
public class AppManager extends Panel
	{
		public function AppManager()
		{
 
		}
 
		override protected function initialize():void
		{
			new MetalWorksMobileTheme();
 
			this.layout = new AnchorLayout();
 
			var layoutData:AnchorLayoutData = new AnchorLayoutData();
			layoutData.horizontalCenter = 0;
			layoutData.verticalCenter = 0;
 
			var textInput:TextInput = new TextInput();
			this.addChild( textInput );
		}
 
	}
```

**Comment by joshtynjala** [comment]
Feathers does not change the full screen state in any way. The status bar is fully controlled by Adobe AIR, so this is an AIR bug.

**Comment by sebastiantoro** [comment]
Do you know any way to bypass this issue?

**Comment by joshtynjala** [comment]
You could try setting the displayState of the nativeStage to StageDisplayState.FULL_SCREEN_INTERACTIVE on focus out.

What Android device are you using for testing? Is it happening on other Android devices too?

**Comment by sebastiantoro** [comment]
>You could try setting the displayState of the nativeStage to StageDisplayState.FULL_SCREEN_INTERACTIVE on focus out.

When the TextInput is created the ´Starling.current.nativeStage.displayState´ is StageDisplayState.FULL_SCREEN_INTERACTIVE. It has the exact same value in focus in and in focus out. Is that what you meant?

>What Android device are you using for testing? Is it happening on other Android devices too?

The bug occurred on the following devices:
- Tablet: Samsung Galaxy Tab 4 7" (SM-T230) - Android 4.4.2
- Tablet: Samsung Galaxy Tab A 7" (SM-T280) - Android 5.1.1
- Tablet: Samsung Galaxy Note 10.1" (SM-P600) - Android 4.4.2
The bug did not occurred on the following devices:
- Tablet: Samsung Galaxy Tab 3 7" (SM-T210) - Android 4.4.2
The bug occurred occurred only sometimes on the following devices:
- Phone: Huawei G Play mini (CHC-U03) - Android 4.4.2

**Comment by joshtynjala** [comment]
I was assuming that nativeStage.displayState changed to StageDisplayState.NORMAL when the status bar appeared. If that's not the case, then there may not be much you can do. You could possibly try something like this:

```
Starling.current.nativeStage.displayState = StageDisplayState.NORMAL;
Starling.current.nativeStage.displayState = StageDisplayState.FULL_SCREEN_INTERACTIVE;
```

If that doesn't work, then there's probably nothing you can do except to file a bug report with Adobe.

**Comment by sebastiantoro** [comment]
This problewas filed to Adobe and they "solved" it on AIR 24. In that version, the status bar dissapeared when you focus out from the TextInput but when you focus in, the screen moves up and down very erratically. 

On version 25 of the AIR SDK the problem is completely solved at leat on Samsung Galaxy Tab 4 7" (SM-T230)  I didn't test the other tablets because I started testing AIR 26 on the rest. I tested version 26 on SM-T230, SM-T280 and SM-T210 and all work like a charm (didn't test on the rest).
